Source/WebCore/dom/FragmentDirectiveRangeFinder.cpp

```diff
@@ -151,12 +151,24 @@ static std::optional<SimpleRange> findRangeFromNodeList(const String& query, con
     StringBuilder searchBufferBuilder;
     for (auto& node : nodes)
         searchBufferBuilder.append(node->data());
-    // FIXME: try to use SearchBuffer in TextIterator.h instead.
     searchBuffer = searchBufferBuilder.toString();
     
     searchBuffer = foldQuoteMarks(searchBuffer);
     auto foldedQuery = foldQuoteMarks(query);
     
+    // FIXME: add quote folding to TextIterator instead of leaving it here?
+    FindOptions options = { CaseInsensitive, DoNotRevealSelection };
+
+    if (wordStartBounded == WordBounded::Yes)
+        options.add(AtWordStarts);
+    if (wordEndBounded == WordBounded::Yes)
+        options.add(AtWordEnds);
+
+    auto foundText = findPlainText(searchRange, foldedQuery, options);
+
+    if (!foundText.collapsed())
+        return foundText;
+
```

Comment:
So we fallback to just matching text when there is no match?
We should probably mention that in the commit message.

Reply:
Yes, and good point!